packages/ui/src/components/surfaces/accordion/index.tsx

```diff
@@ -35,14 +44,22 @@ export const Accordion = ({
 }: AccordionProps): ReactElement => {
     const [internalExpanded, setInternalExpanded] = useState<boolean>(false);
 
+    const childSteps = useMemo(
+        () => React.Children.toArray(children),
+        [children]
+    );
     const isControlled = useMemo(() => expanded !== undefined, [expanded]);
     const isExpanded = useMemo(
         () => (isControlled ? !!expanded : internalExpanded),
         [isControlled, expanded, internalExpanded]
     );
 
-    const summaryChildren = children[0];
-    const detailsChildren = children[1];
+    const summaryChildren = childSteps.find((child) =>
```

Comment:
We've used useMemo above and 2 loops here. Wondering if it makes sense to have a reduce on the children in the `useMemo` to just do one single loop and have both the accordion summary and details.

Reply:
Do you mean something like this?
```tsx
const { summaryChildren, detailsChildren } = React.Children.toArray(
    children
).reduce<{
    summaryChildren: ReactNode;
    detailsChildren: ReactNode;
}>(
    (previous, current) => {
        if (matchChildByType(current, AccordionSummary)) {
            return {
                ...previous,
                summaryChildren: current,
            };
        }
        if (matchChildByType(current, AccordionDetails)) {
            return {
                ...previous,
                detailsChildren: current,
            };
        }

        return previous;
    },
    { summaryChildren: null, detailsChildren: null }
);
````